src/payment/strategies/zip/zip-payment-strategy.ts

```diff
@@ -0,0 +1,145 @@
+            });
+    }
+
+    deinitialize(options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        this._paymentMethod = undefined;
+        this._zipClient = undefined;
+
+        return Promise.resolve(this._store.getState());
+    }
+
+    execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        if (!this._paymentMethod) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        return this._displayWallet();
+    }
+
+    finalize(options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        return Promise.reject(new OrderFinalizationNotRequiredError());
+    }
+
+    private _displayWallet(): Promise<InternalCheckoutSelectors> {
+        return this._store.dispatch(this._paymentStrategyActionCreator.widgetInteraction(() => {
+            this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(this._methodId))
+                .then(state => {
+                    if (!this._zipClient) {
+                        throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+                    }
+
+                    this._zipClient.Checkout.init(this._getZipPayload());
+                });
+
+            // Lightbox closing event listener
+            return new Promise((resolve, reject) => {
+                this._lightboxEvents$
+                    .pipe(take(1))
+                    .subscribe((event: { type: ZipModalEvent }) => {
+                        if (event.type === ZipModalEvent.CancelCheckout) {
+                            reject(new PaymentMethodCancelledError());
+                        }
+                    });
+            });
+        }, { methodId: this._methodId }), { queueId: 'widgetInteraction' });
+    }
+
+    private _getZipPayload(): ZipPayload {
+        return {
+            onComplete: this._handleResponse,
+            onCheckout: this._fetchPayload,
+        };
+    }
+
+    @bind
+    private _fetchPayload(resolve: (response: ZipPostResponse) => void): void {
+        this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(this._methodId))
+            .then(state => {
+                const paymentMethod = state.paymentMethods.getPaymentMethod(this._methodId);
+
+                if (!paymentMethod || !paymentMethod.clientToken) {
+                    throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+                }
+
+                resolve(JSON.parse(paymentMethod.clientToken));
+            });
+    }
+
+    @bind
+    private _handleResponse(response: ZipResponse): void {
+        if (response.state === ZipModalEvent.CancelCheckout) {
+            this._lightboxEvents$.next({ type: ZipModalEvent.CancelCheckout });
+
+            return;
+        }
+        const { state, checkoutId } = response;
+
+        this._requestSender.post('checkout.php', {
```

Comment:
Please double check with checkout team if this is a valid way to set checkout data.